mooncake-store/src/client.cpp

```diff
@@ -105,6 +109,38 @@ static std::vector<std::string> get_auto_discover_filters(bool auto_discover) {
     return whitelst_filters;
 }
 
+ErrorCode Client::ConnectToMaster(const std::string& master_entries, bool enable_ha) {
+    if (enable_ha) {
+        // Get master address from ETCD
+        auto err = master_view_helper_.ConnectToEtcd(master_entries);
+        if (err != ErrorCode::OK) {
+            LOG(ERROR) << "Failed to connect to etcd";
+            return err;
+        }
+        std::string master_address;
+        ViewVersionId master_version = 0;
+        err = master_view_helper_.GetMasterView(master_address, master_version);
+        if (err != ErrorCode::OK) {
+            LOG(ERROR) << "Failed to get master address";
+            return err;
+        }
+
+        err = master_client_.Connect(master_address);
+        if (err != ErrorCode::OK) {
+            LOG(ERROR) << "Failed to connect to master";
+            return err;
+        }
+
+        // Start Ping thread to monitor master view changes and remount segments if needed
+        ping_running_ = true;
+        ping_thread_ = std::thread(&Client::PingThreadFunc, this, master_version);
```

Comment:
This can be a heartbeat thread also, for master, if the heartbeat lost for a while, master can unmount the segment from this client.

Reply:
Do you suppose to make this thread to be daemon thread?

Reply:
Yes. These regular Pings can be used by master to detect the liveness of clients. We will implement this feature in the next PR.